audit/conn/conn.go

```diff
@@ -0,0 +1,33 @@
+// Package conn provides interfaces for audit server connections.
+package conn
+
+import (
+	"context"
+
+	"github.com/microsoft/go-otel-audit/audit/msgs"
+)
+
+//go:generate stringer -type=Type -linecomment
+
+// Type represents the type of the audit connection.
+type Type uint8
+
+const (
+	TypeUnknown      Type = 0 // Unknown
+	TypeNoOP         Type = 1 // NoOP
+	TypeDomainSocket Type = 2 // UnixDomainSocket
+	TypeTCP          Type = 3 // TCP
+)
+
+// Audit represents a connection to a remote audit server.
+// This is internal only and implemenation by external pacakges is not supported.
+type Audit interface {
```

Comment:
What are the pros/cons of just putting this into `internal`?

Reply:
Pros: none that I can think of, since it can only be developed within this package due to private() method.
Cons: *slightly* harder to figure out that all the clients are implementing this.

The only thing about internal is it is a solution to a problem in which we need "visibility" rules like bazel. But that can lead to some unfortunate consequences.  "internal" has a radius of importability and if you exceed that you end up having to restructure to get at a package (or use linker directives).  If I moved this, I don't think we'd trip it, but I'm not sure its worth it.  

If you've got a reason to move it, I'll yank it.